connect-rest-extension/src/main/java/io/debezium/kcdrestextension/DebeziumResource.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.kcdrestextension;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.ServletContext;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.apache.kafka.common.utils.AppInfoParser;
+import org.apache.kafka.connect.health.ConnectClusterState;
+import org.apache.kafka.connect.runtime.Herder;
+import org.apache.kafka.connect.runtime.health.ConnectClusterStateImpl;
+import org.apache.kafka.connect.runtime.isolation.PluginDesc;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.apache.kafka.connect.transforms.predicates.HasHeaderKey;
+import org.apache.kafka.connect.transforms.predicates.RecordIsTombstone;
+import org.apache.kafka.connect.transforms.predicates.TopicNameMatches;
+import org.apache.maven.artifact.versioning.ComparableVersion;
```

Comment:
Let's avoid adding this dependency solely for this single purpose. Either create our own class for it, or see whether we can use `java.lang.Runtime$Version`, available since Java 9. We can require Java 11 as the minimum baseline for this extension.

Reply:
I wanted to, but I somehow thought things in the main repo inheriting from the parent pom must be Java 8. If it's possible I will do the change, sure!

Reply:
I forgot on my previous comment that Kafka Connect needs to run on Java 8 and we need to keep Java 8 compatibility. I have a custom class in the initial repo. I can bring back that class for the version comparison, but I feel more confident with using the original gradle or maven code for version handling to ensure predictable behavior as I am handling maven or gradle version strings here. wdyt?

Reply:
> I forgot on my previous comment that Kafka Connect needs to run on Java 8 and we need to keep Java 8 compatibility

Why is that? The container images [have Java 11|https://github.com/debezium/docker-images/blob/master/base/latest/Dockerfile#L7].

Reply:
https://kafka.apache.org/28/documentation.html#java states compatibility with Java 8 and Java 11.

Reply:
We build this for usage via the Debezium Kafka Connect image, which is based on Java 11. I don't expect any other user to deploy this extension. And if they do, requiring them to run on Java 11 is a fair expectation.